src/core/qgis.h

```diff
@@ -1804,6 +1804,44 @@ class CORE_EXPORT Qgis
     };
     Q_ENUM( CoordinateOrder )
 
+    /**
+     * Available identifier string types for representing coordinate reference systems
+     *
+     * \note Prior to QGIS 3.36 this was available as QgsCoordinateReferenceSystem::IdentifierType
+     *
+     * \since QGIS 3.36
+     */
+    enum class CrsIdentifierType SIP_MONKEYPATCH_SCOPEENUM_UNNEST( QgsCoordinateReferenceSystem, IdentifierType ) : int
+      {
+      ShortString, //!< A heavily abbreviated string, for use when a compact representation is required
+      MediumString, //!< A medium-length string, recommended for general purpose use
+      FullString, //!< Full definition -- possibly a very lengthy string, e.g. with no truncation of custom WKT definitions
+    };
+    Q_ENUM( CrsIdentifierType )
+
+    /**
+     * Coordinate reference system WKT formatting variants.
+     *
+     * \note Prior to QGIS 3.36 this was available as QgsCoordinateReferenceSystem::WktVariant
+     *
+     * \since QGIS 3.36
+     */
+    enum class CrsWktVariant SIP_MONKEYPATCH_SCOPEENUM_UNNEST( QgsCoordinateReferenceSystem, WktVariant ) : int
+      {
+      Wkt1GDAL SIP_MONKEYPATCH_COMPAT_NAME( WKT1_GDAL ), //!< WKT1 as traditionally output by GDAL, deriving from OGC 01-009. A notable departure from WKT1_GDAL with respect to OGC 01-009 is that in WKT1_GDAL, the unit of the PRIMEM value is always degrees.
+      Wkt1ESRI SIP_MONKEYPATCH_COMPAT_NAME( WKT1_ESRI ), //!< WKT1 as traditionally output by ESRI software, deriving from OGC 99-049.
+      Wkt2_2015 SIP_MONKEYPATCH_COMPAT_NAME( WKT2_2015 ), //!< Full WKT2 string, conforming to ISO 19162:2015(E) / OGC 12-063r5 with all possible nodes and new keyword names.
+      Wkt2_2015Simplified SIP_MONKEYPATCH_COMPAT_NAME( WKT2_2015_SIMPLIFIED ), //!< Same as WKT2_2015 with the following exceptions: UNIT keyword used. ID node only on top element. No ORDER element in AXIS element. PRIMEM node omitted if it is Greenwich.  ELLIPSOID.UNIT node omitted if it is UnitOfMeasure::METRE. PARAMETER.UNIT / PRIMEM.UNIT omitted if same as AXIS. AXIS.UNIT omitted and replaced by a common GEODCRS.UNIT if they are all the same on all axis.
+      Wkt2_2018 SIP_MONKEYPATCH_COMPAT_NAME( WKT2_2018 ), //!< Alias for WKT2_2019
+      Wkt2_2018Simplified SIP_MONKEYPATCH_COMPAT_NAME( WKT2_2018_SIMPLIFIED ), //!< Alias for WKT2_2019_SIMPLIFIED
```

Comment:
Is there a point in providing 2018 constants ? They were just transitional

Reply:
@rouault 

It's just maintaining API compatibility -- the original enum was first introduced in 2018 and had JUST the 2018 constants. Then we added the 2019 aliases to them later, and now we have to drag everything forward. 

That said, I could potentially remove them from c++ and monkey patch them into PyQGIS only. I'll do that...!

Reply:
Done